File: WIS/source/WIS_Prototype/WIS_PrototypeAPI/Data/DTOs/IntakeReport.cs
//************************************************
// Filename: IntakeReport.cs
// Purpose: Defines an object which is used to transport Report data from the database.

using Microsoft.EntityFrameworkCore;
using WIS_PrototypeAPI.Data;
using WIS_PrototypeAPI.Data.Models;

namespace WIS_PrototypeAPI.Data.DTOs
{
	[Keyless]
	public class IntakeReport
	{
		// Commodity Type
		public int CommodityTypeIdLink { get; set; }
		//public string? CommodityTypeName { get; set; } = null;

		// Commodity Variety
		public long? CommodityVarietyIdLink { get; set; }
		//public string? CommodityVarietyName { get; set; } = null;

		// Producer number
		public int? ProducerIdLink { get; set; }
		// Producer name
		public string? ProducerName { get; set; } = null;

		// Lot number
		public long LotIdLink { get; set; }

		// Weight Sheet number
		public long WeightsheetId { get; set; }

		// lot is open or closed
		public DateTime? EndDate { get; set; } = null;

		// Landlord
		public string? Landlord { get; set; } = null;

		// FSA farm number
		public string? FarmNumber { get; set; } = null;

		// Net (lbs)
		public long NetWeightLbs { get; set; }
	}
}

File: WIS/source/WIS_Prototype/WIS_Prototype/src/app/report/intake-report.ts
export class IntakeReport {
  public commodityTypeIdLink: number;
  public commodityTypeName: string;
  public commodityVarietyIdLink?: bigint;
  public commodityVarietyName?: string;
  public producerIdLink: number;
  public producerName: string;
  public lotIdLink: number;
  public weightsheetId: bigint;
 // public dateLotOpened: Date | null;
  public endDate: Date | null;
  public landlord: string | null;
  public farmNumber: number | null;
  public netWeightLbs: number;
  public netUom: number;
  public isClosedString: string | null;

  constructor() {
    this.commodityTypeIdLink = 0;
    this.commodityTypeName = "";
    this.commodityVarietyIdLink = BigInt(0);
    this.commodityVarietyName = "";
    this.producerIdLink = 0;
    this.producerName = "";
    this.lotIdLink = 0;
    this.weightsheetId = BigInt(0);
    //this.dateLotOpened = null;
    this.endDate = null;
    this.landlord = null;
    this.farmNumber = null;
    this.netWeightLbs = 0;
    this.netUom = 0;
    this.isClosedString = null;
  }

  // Populates a field that allows for easier display on the table.
  SetClosedToString(): void {
    if (this.endDate == null) {
      this.isClosedString = "Open";
    }
    else {
      this.isClosedString = "Closed";
    }
  }

  AsBushel(): number {
    if (this.netWeightLbs != null) {
      return (this.netWeightLbs / 60);
    }
    else {
      return -1;
    }
  }
}
